Fix Wifi connection issues - public networks and password handling

Chirag8405 opened the pull request:
# Fix WiFi Connection Issues - Public Networks & Password Handling

##  Fixes Issues
- Closes #58  - Unable to connect to public wifi from sleex settings

##  Changes Made

### Core Fixes
- **Public WiFi Support**: Enable connection to open networks without password prompts
- **Incorrect Password Detection**: Implement specific error messages for authentication failures  
- **WiFi Toggle Fix**: Restore proper WiFi enable/disable functionality
- **Enhanced Error Handling**: Fast authentication failure detection (1s vs 3s+ previously)

### Technical Implementation
- **NetworkManager Integration**: Enhanced device state monitoring with specific error reasons
- **Dynamic Security Detection**: Proper handling of networks changing security settings
- **UI/UX Improvements**: Clean error message positioning without layout conflicts
- **ScriptModel Compliance**: Fix invalidate() errors with proper refresh triggers

### Files Modified
- `src/share/sleex/plugins/src/Sleex/services/network.cpp` - Core NetworkManager logic
- `src/share/sleex/modules/settings/Wifi.qml` - UI enhancements and error handling

##  Testing Done
- [x] Public hotspot connection (no password required)
- [x] Secured network connection (WPA2/WPA3 with password)
- [x] Incorrect password detection and user feedback
- [x] Network security change detection (open ↔ protected)
- [x] WiFi toggle enable/disable functionality  
- [x] Error message display and UI stability

##  Backward Compatibility
- ✅ No breaking changes to existing API
- ✅ All existing WiFi functionality preserved
- ✅ Enhanced error handling is additive only

##  Additional Notes
This PR resolves the core WiFi connectivity issues that prevented users from connecting to public networks while maintaining all existing functionality for secured networks.

Chirag8405 committed:
Fix Wifi connection issues - public networks and password handling

- Fix public WiFi connection without password requirement (#58)
- Implement proper incorrect password detection and error messages
- Fix WiFi toggle functionality
- Add enhanced NetworkManager state detection
- Improve error message positioning and UI aesthetics

LeVraiArdox commented on src/share/sleex/plugins/src/Sleex/services/network.cpp:
```diff
@@ -256,21 +308,65 @@ void Network::rescanWifi() {
+        // Network security changed from open to protected - recreate connection
+        
+        GError *deleteError = nullptr;
+        gboolean deleteResult = nm_remote_connection_delete(connection, nullptr, &deleteError);
```
`nm_remote_connection_delete` is deprecated. Can you use something else such as `nm_remote_connection_delete_async` ?

LeVraiArdox commented on src/share/sleex/modules/settings/Wifi.qml:
```diff
@@ -71,39 +139,135 @@ ContentPage {
+
+            MaterialSymbol {
+                Layout.alignment: Qt.AlignHCenter
+                text: "wifi_off"
```
use the "signal_wifi_off" icon, it looks better.

LeVraiArdox commented on src/share/sleex/modules/settings/Wifi.qml:
```diff
@@ -71,39 +139,135 @@ ContentPage {
+        }
+
+        // Connection error display
+        Rectangle {
```
It would be better to put that as a red text under the current entry.

LeVraiArdox commented on src/share/sleex/modules/settings/Wifi.qml:
```diff
@@ -190,38 +362,93 @@ ContentPage {
-                                        networkItem.expanded = false
-                                    }
+                            // Forget network button for known networks
+                            RippleButton {
```
Here, do not put this button here.

LeVraiArdox commented on src/share/sleex/modules/settings/Wifi.qml:
I don't really like the design of this. The forget button is right next to two other important buttons, that's an UX inconsistency. Also, the wifi input is now visible even when the network is known. And What you can do is keep the previous layout but use RippleButtonWithIcon for the forget button. As a bonus, do the same for the expand.

LeVraiArdox reviewed (changes requested):
Told you everything is needed. It's almost all visual stuff. The backend works fine. Good job!

Chirag8405 committed:
Improved UI Consistency

Chirag8405 committed:
Fix WiFi UI design and UX inconsistencies

- Change WiFi off icon to "signal_wifi_off" for better visual clarity
- Restore original layout structure to maintain UX consistency
- Fix password input visibility to only show for unknown networks
- Convert forget button to use RippleButtonWithIcon design pattern
- Convert expand arrow to use RippleButtonWithIcon for consistency
- Remove forget button from inline position to prevent accidental clicks

Chirag8405 commented:
@LeVraiArdox  I have made the necessary changes as adviced by you.

LeVraiArdox commented on src/share/sleex/modules/settings/Wifi.qml:
```diff
@@ -173,55 +320,111 @@ ContentPage {
+                            RippleButtonWithIcon {
+                                visible: networkItem.modelData?.isSecure || false
+                                materialIcon: networkItem.expanded ? "keyboard_arrow_up" : "keyboard_arrow_down"
+                                mainText: ""
```
Maybe "expand" instead of blank ?

Chirag8405 replied:
I kept it blank as it displays that text beside that icon.

LeVraiArdox replied:
Yeah, I think it looks better with the text

Chirag8405 replied:
Ok will add that, but I was thinking to add text to be revealed when hovered over the arrow

Chirag8405 replied:
<img width="941" height="185" alt="image" src="https://github.com/user-attachments/assets/d940d711-729c-4d43-bfd7-564e0a9798aa" />

<img width="901" height="194" alt="image" src="https://github.com/user-attachments/assets/6a2bf18f-800c-477b-b40e-907e4bb7b52a" />

The second image is of hover to reveal.

LeVraiArdox replied:
On the second image, you can see there is a weird right padding on the button. In that case, the normal Ripple Button is better

Chirag8405 replied:
<img width="907" height="196" alt="image" src="https://github.com/user-attachments/assets/51684cee-2098-49a8-99eb-e09397d34d67" />

Used normal Ripple Button. If this works, then i can push the final commit. I have solved the configuring connection issue and removed the fill width as you have suggested.

LeVraiArdox replied:
Good

LeVraiArdox commented on src/share/sleex/modules/settings/Wifi.qml:
```diff
@@ -388,6 +555,21 @@ ContentPage {
+
+                                // Forget button for known networks, separated at bottom
+                                RippleButtonWithIcon {
+                                    Layout.fillWidth: true
```
Here. Remove this one, it looks ugly when fillwidthed (that's not a word but you get it)

LeVraiArdox reviewed (changes requested):
Minor changes are demanded. But it's clearly better than the previous time!

Also, look at this

<img width="512" height="117" alt="image" src="https://github.com/user-attachments/assets/6e65b62c-1079-4dee-a6c9-1490f238ff4b" />

When we enter incorrect password, it says incorrect password for a second, but then it show this. Possibly the next message. Can you fix that ? If not, it's okay, we can keep it like that.

Chirag8405 committed:
Fix WiFi authentication error handling and UI consistency

- Prevent duplicate error messages overriding "Incorrect password"
- Clear error tracking on successful connections
- Improve expand button UI consistency

LeVraiArdox reviewed (approved):
LGTM

Bonus if you want, the wifi icon seems to be marked as no signal when you are not connected to them. I can do it myself, but if you want to do it I can let you do it. 

Else, we can merge, it's all good.

Chirag8405 committed:
fix: WiFi icon shows signal when enabled but not connected, add security property

Chirag8405 commented:
I have added the wifi signal icons according to their strengths. Also previously the expanded menu had security which always showed up as undefined, so fixed that too.

LeVraiArdox commented:
Thank you for your contribution, it's really helpful!
